lib/facility.rb

```diff
@@ -1,14 +1,64 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
 
-  def initialize(name, address, phone)
+  attr_reader :name,
+              :address,
+              :phone,
+              :services,
+              :registered_vehicles,
+              :collected_fees
+
+
+  #change to look for keyword instead of positional 
+  def initialize(name:, address:, phone:)
     @name = name
     @address = address
     @phone = phone
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  #each time a vehicle is registered, a fee is collected based on vehicle type
+  def register_vehicle(vehicle)
+   if @services.include?('Vehicle Registration')
+    vehicle.register 
+    if vehicle.antique?
+      @collected_fees += 25
+    elsif vehicle.electric_vehicle?
+      @collected_fees += 200
+    else
+      @collected_fees += 100
+    end
+    @registered_vehicles << vehicle
+   end
+  end
```

Comment:
Nice use of your conditionals here to check if the service is available first, and then collect fees accordingly.